Add ajax navigation to cfr changes toc.

jmcarp opened the pull request:
There's more to do here--I'm guessing we'll want to open the preamble toc panel when viewing a preamble section, and the cfr changes toc panel when viewing cfr changes, for example. We might also want to refactor `DrawerView` to hard-code less stuff, if possible (why are we instantiating views with no corresponding dom elements?). Happy to add more to this patch, or continue work elsewhere.

@cmc333333: there's more ugliness here around building section ids for the cfr changes toc. I can modify the toc generator to add prefixed fields if you think that's the right approach.

cmc333333 commented on regulations/static/regulations/js/source/views/drawer/drawer-view.js:
```diff
@@ -27,6 +27,11 @@ var DrawerView = Backbone.View.extend({
+        var $tocSecondary = $('table-of-contents-secondary');
+        if ($tocSecondary.length) {
+          this.childViews['table-of-contents-secondary'] = new TOCView({el: $tocSecondary});
+        }
```
Nice reuse!

cmc333333 commented on regulations/templates/regulations/preamble-chrome.html:
```diff
@@ -10,7 +10,7 @@
   </a>
 </li>
-<li><a href="#timeline" id="timeline-link" class="toc-nav-link" title="CFR Changes">CFR</a></li>
+<li><a href="#table-of-contents-secondary" id="timeline-link" class="toc-nav-link" title="CFR Changes">CFR</a></li>
```
Probably want a different id here; I wonder what it keys?

cmc333333 commented on regulations/templates/regulations/preamble-chrome.html:
```diff
@@ -65,13 +65,10 @@ <h3 class="subpart-heading">{{cfr_part_toc.title}} CFR {{cfr_part_toc.part}}</h3
       </ol>
-      {% with toc_items=TOC %}
-        {% include "regulations/toc.html" %}
-      {% endwith %}
```
Thank you, forgot to remove that before committing.

cmc333333 commented on regulations/templates/regulations/preamble-chrome.html:
```diff
@@ -53,7 +53,7 @@ <h3 class="toc-type">Table of Contents</h3>
   @TODO: "timeline" is not an accurate description here, but the JS has a
   whitelist which needs to be followed. We can follow up to generalize the JS
```
No longer need this comment, hey?

cmc333333 commented:
Thanks for making this work!

cmc333333 commented:
> Happy to add more to this patch, or continue work elsewhere.
If you see a quick change, go for it, but I'm content to merge as is, aside from the minor comments above

> I can modify the toc generator to add prefixed fields if you think that's the right approach.
Seems worthwhile -- I think we'll probably have a similar generator for the preamble ToC (walking the tree in the template is kind of silly)

jmcarp commented:
@cmc333333: I cleaned up comments, pushed the section id logic to `CFRChangeToC`, and updated to load the appropriate toc on page load and comment edit.

cmc333333 commented on regulations/static/regulations/js/source/app-init.js:
```diff
@@ -25,13 +25,13 @@ Backbone.$ = $;
+        var header = new HeaderView();  // Header before Drawer as Drawer sends Header events
+        var drawer = new DrawerView();
+        var main = new MainView();
+        var sidebar = new SidebarView();
```
Minor, but I think this breaks existing style conventions; why not use the comma approach?

jmcarp replied:
That's just me trying to inject my js style preferences, although I borrowed them from airbnb: https://github.com/airbnb/javascript#variables. I think the frontend guild recommends the airbnb guide. Anyway, I'm happy to roll that back if you prefer.

cmc333333 replied:
NBD. Opened #204 for consistency

cmc333333 commented on regulations/static/regulations/js/source/views/drawer/drawer-tabs-view.js:
```diff
@@ -17,6 +17,7 @@ var DrawerTabsView = Backbone.View.extend({
         'table-of-contents': '#menu-link',
+        'table-of-contents-secondary': '#table-of-contents-secondary-link',
         'timeline': '#timeline-link',
         'search': '#search-link'
```
We can delay this for later, but can this be generalized by naming these more consistently? Just swapping `#menu-link` for `#table-of-contents-link`?

jmcarp replied:
I was also thinking about putting these relationships on the dom so the js doesn't have to know about all possible pairs of link and section ids.

cmc333333 replied:
<3 declarative

cmc333333 commented on regulations/static/regulations/js/source/views/main/preamble-view.js:
```diff
@@ -38,6 +43,7 @@ var PreambleView = ChildView.extend({
     this.listenTo(MainEvents, 'paragraph:active', this.handleParagraphActive);
 
     CommentEvents.trigger('comment:readTabOpen');
+    DrawerEvents.trigger('pane:init', type === 'preamble' ? 'table-of-contents' : 'table-of-contents-secondary');
```
I could imagine DrawerEvents like these being driven by the DOM, too, rather than trying to parse out the state in JS.
